app/src/lib/editors/win32.ts

```diff
@@ -0,0 +1,172 @@
+import * as path from 'path'
+import * as Registry from 'winreg'
+
+import {
+  FoundEditor,
+  AtomLabel,
+  VisualStudioCodeLabel,
+  SublimeTextLabel,
+} from './shared'
+
+/**
+ * Find the Atom executable shim using the install information from the
+ * registry.
+ */
+function findAtomExecutable(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const regKey = new Registry({
+      hive: Registry.HKCU,
+      key: '\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\atom',
+    })
+
+    regKey.values((err, items) => {
+      if (err) {
+        reject(err)
+        return
+      }
+
+      let displayName = ''
+      let publisher = ''
+      let installLocation = ''
+
+      for (const item of items) {
+        if (item.name === 'DisplayName') {
+          displayName = item.value
+        } else if (item.name === 'Publisher') {
+          publisher = item.value
+        } else if (item.name === 'InstallLocation') {
+          installLocation = item.value
+        }
+      }
+
+      if (displayName === 'Atom' && publisher === 'GitHub Inc.') {
+        resolve(path.join(installLocation, 'bin', 'atom.cmd'))
+        return
+      }
+
+      reject('Registry entry does not match expected settings for Atom')
+    })
+  })
+}
+
+/**
+ * Find the Visual Studio Code executable shim using the install information
+ * from the registry.
+ */
+function findCodeExecutable(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const regKey = new Registry({
+      hive: Registry.HKLM,
+      key:
+        '\\SOFTWARE\\WOW6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\{F8A2A208-72B3-4D61-95FC-8A65D340689B}_is1',
```

Comment:
I tested three different installs of VS Code, this GUID is consistent across all of the installs.